### PR TITLE
Stm32mp1 i2c arduino

### PR DESCRIPTION
**Pull request by arnopo**
Add support of I2C on arduino connector fro stm32mp157c-dk2.

Tested via samples/shields/x_nucleo_iks01a2.

This pull request replace the #17044 that is now deprecated and can be closed:
- rebase of the commits on master.
- update of the west.yaml after merge of the I2C support in hal_stm32 module

### CODE REVIEW

**Review by ydamigos — APPROVED** [review]

File: drivers/i2c/i2c_ll_stm32.c [hunk 1]
```diff
@@ -424,3 +424,48 @@ static void i2c_stm32_irq_config_func_4(struct device *dev)
 #endif
 
 #endif /* CONFIG_I2C_4 */
+
+
+#ifdef CONFIG_I2C_5
```
Comment (erwango) [inline drivers/i2c/i2c_ll_stm32.c]: Conflict with https://github.com/zephyrproject-rtos/zephyr/pull/17818 to be taken care

File: drivers/i2c/Kconfig.stm32 [hunk 2]
```diff
@@ -23,13 +23,14 @@ config I2C_STM32_V1
 
 config I2C_STM32_V2
 	bool
-	depends on SOC_SERIES_STM32F0X || SOC_SERIES_STM32F3X || SOC_SERIES_STM32F7X || SOC_SERIES_STM32L0X || SOC_SERIES_STM32L4X || SOC_SERIES_STM32WBX
+	depends on SOC_SERIES_STM32F0X || SOC_SERIES_STM32F3X || SOC_SERIES_STM32F7X || SOC_SERIES_STM32L0X || SOC_SERIES_STM32L4X || SOC_SERIES_STM32WBX || SOC_SERIES_STM32MP1X
```
Comment (erwango) [inline drivers/i2c/Kconfig.stm32]: This change does not match the commit message header

File: soc/arm/st_stm32/stm32mp1/Kconfig.defconfig.series [hunk 3]
```diff
@@ -47,4 +47,11 @@ config IPM_STM32_IPCC
 
 endif # IPM
 
+if I2C_STM32
```
Comment (erwango) [inline soc/arm/st_stm32/stm32mp1/Kconfig.defconfig.series]: This does not match the commit header message

File: soc/arm/st_stm32/stm32mp1/soc.h [hunk 4]
```diff
@@ -50,6 +50,10 @@
 #include <stm32mp1xx_ll_ipcc.h>
 #endif
 
+#ifdef CONFIG_I2C
```
Comment (erwango) [inline soc/arm/st_stm32/stm32mp1/soc.h]: ditto

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/i2c/Kconfig.stm32 (hunk 2 above)
Comment (arnop2) [reply drivers/i2c/Kconfig.stm32]: commit message updated

File: soc/arm/st_stm32/stm32mp1/Kconfig.defconfig.series (hunk 3 above)
Comment (arnop2) [reply soc/arm/st_stm32/stm32mp1/Kconfig.defconfig.series]: commit message and subject updated

**Review by erwango — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks.

**Comment by erwango** [comment]
@arnop2 , please rebase to fix CI

**Comment by arnop2** [comment]
> @arnop2 , please rebase to fix CI

done

**Comment by erwango** [comment]
@galak, can we get this one in ?

**Pull request merged**